src/mixins/filing-template-mixin.ts

```diff
@@ -559,129 +567,137 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
   //
 
   /**
-   * Parses a draft IA/change/registration correction filing into the store.
+   * Parses a draft IA/change/registration Correction filing into the store.
    * @param filing the correction filing
+   * @param entitySnapshot the latest entity snapshot
    */
-  parseCorrectionFiling (filing: CorrectionFilingIF): void {
-    // Store business information
+  parseCorrectionFiling (filing: CorrectionFilingIF, entitySnapshot: EntitySnapshotIF = null): void {
+    // store Entity Snapshot
+    this.setEntitySnapshot(entitySnapshot)
+
+    // store Business Information
     this.setBusinessInformation(filing.business)
 
-    // Store name request
-    if (filing.incorporationApplication) this.setNameRequest(filing.incorporationApplication.nameRequest)
-    if (filing.changeofRegistration) this.setNameRequest(filing.changeofRegistration.nameRequest)
-    if (filing.registration) this.setNameRequest(filing.registration.nameRequest)
-
-    // Store name translations
-    // NB: The first time (when we initiate a correction), the `oldName` and `action` props are not
-    //     available in the api response, which creates an issue of not having these props in store.
-    //     Due to missing props, change event was not triggering if the action value is changed (at
-    //     the time of Delete there is no other prop change except action). To handle this scenario,
-    //     this structure needs to be kept.
-    if (filing.incorporationApplication) { // *** FUTURE: expand for other filing types
+    // store Correction Information
+    this.setCorrectionInformation(filing.correction)
+
+    // store Name Request
+    if (filing.incorporationApplication?.nameRequest) {
+      this.setNameRequest(filing.incorporationApplication.nameRequest)
```

Comment:
This kind of code will set the Name Request if the object exists in the IA, otherwise it leaves the store with default values. The code below is similar.